Data: azurerm_kubernetes_cluster crash on v3.106

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.5.6

### AzureRM Provider Version

3.106.0

### Affected Resource(s)/Data Source(s)

data.azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
variable "aks_name" {
  description = "Name of the AKS device to deploy to"
  type        = string
}

variable "aks_resource_group_name" {
  description = "Name of the resource group containing the AKS device"
  type        = string
}

data "azurerm_kubernetes_cluster" "aks" {
  name                = var.aks_name
  resource_group_name = var.aks_resource_group_name
}
```


### Debug Output/Panic Output

```shell
2024-05-31T08:49:52Z logger.go:66: 
2024-05-31T08:49:52Z logger.go:66: Planning failed. Terraform encountered an error while generating this plan.
2024-05-31T08:49:52Z logger.go:66: 
2024-05-31T08:49:52Z logger.go:66: 
2024-05-31T08:49:52Z logger.go:66: Error: Plugin did not respond
2024-05-31T08:49:52Z logger.go:66: 
2024-05-31T08:49:52Z logger.go:66:   with data.azurerm_kubernetes_cluster.aks,
2024-05-31T08:49:52Z logger.go:66:   on data.tf line 1, in data "azurerm_kubernetes_cluster" "aks":
2024-05-31T08:49:52Z logger.go:66:    1: data "azurerm_kubernetes_cluster" "aks" {
2024-05-31T08:49:52Z logger.go:66: 
2024-05-31T08:49:52Z logger.go:66: The plugin encountered an error, and failed to respond to the
2024-05-31T08:49:52Z logger.go:66: plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more
2024-05-31T08:49:52Z logger.go:66: details.
2024-05-31T08:49:52Z logger.go:66: 
2024-05-31T08:49:52Z logger.go:66: Stack trace from the terraform-provider-azurerm_v3.106.0_x5 plugin:
2024-05-31T08:49:52Z logger.go:66: 
2024-05-31T08:49:52Z logger.go:66: panic: runtime error: invalid memory address or nil pointer dereference
2024-05-31T08:49:52Z logger.go:66: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5c22ea8]
2024-05-31T08:49:52Z logger.go:66: 
2024-05-31T08:49:52Z logger.go:66: goroutine 121 [running]:
2024-05-31T08:49:52Z logger.go:66: github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.flattenKubernetesClusterDataSourceUpgradeSettings(...)
2024-05-31T08:49:52Z logger.go:66: 	github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/kubernetes_cluster_data_source.go:1483
2024-05-31T08:49:52Z logger.go:66: github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.flattenKubernetesClusterDataSourceAgentPoolProfiles(0xc002491880?)
2024-05-31T08:49:52Z logger.go:66: 	github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/kubernetes_cluster_data_source.go:1241 +0xb88
2024-05-31T08:49:52Z logger.go:66: github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.dataSourceKubernetesClusterRead(0x0?, {0x764db40?, 0xc001d4fb00?})
2024-05-31T08:49:52Z logger.go:66: 	github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/kubernetes_cluster_data_source.go:784 +0x105e
2024-05-31T08:49:52Z logger.go:66: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x8e76da0?, {0x8e76da0?, 0xc002683ef0?}, 0xd?, {0x764db40?, 0xc001d4fb00?})
2024-05-31T08:49:52Z logger.go:66: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:783 +0x163
2024-05-31T08:49:52Z logger.go:66: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc000c808c0, {0x8e76da0, 0xc002683ef0}, 0xc002491700, {0x764db40, 0xc001d4fb00})
2024-05-31T08:49:52Z logger.go:66: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:1015 +0x149
2024-05-31T08:49:52Z logger.go:66: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc0005437b8, {0x8e76da0?, 0xc002683e00?}, 0xc0021e4360)
2024-05-31T08:49:52Z logger.go:66: 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1237 +0x366
2024-05-31T08:49:52Z logger.go:66: github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc001afab40, {0x8e76da0?, 0xc002683650?}, 0xc0014407d0)
2024-05-31T08:49:52Z logger.go:66: 	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:699 +0x3df
2024-05-31T08:49:52Z logger.go:66: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x82a9940?, 0xc001afab40}, {0x8e76da0, 0xc002683650}, 0xc0029964d0, 0x0)
2024-05-31T08:49:52Z logger.go:66: 	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:503 +0x169
2024-05-31T08:49:52Z logger.go:66: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000022000, {0x8ea0720, 0xc0017ea9c0}, 0xc00290f8c0, 0xc0017e9140, 0xe71c008, 0x0)
2024-05-31T08:49:52Z logger.go:66: 	google.golang.org/grpc@v1.58.3/server.go:1374 +0xde7
2024-05-31T08:49:52Z logger.go:66: google.golang.org/grpc.(*Server).handleStream(0xc000022000, {0x8ea0720, 0xc0017ea9c0}, 0xc00290f8c0, 0x0)
2024-05-31T08:49:52Z logger.go:66: 	google.golang.org/grpc@v1.58.3/server.go:1751 +0x9e7
2024-05-31T08:49:52Z logger.go:66: google.golang.org/grpc.(*Server).serveStreams.func1.1()
2024-05-31T08:49:52Z logger.go:66: 	google.golang.org/grpc@v1.58.3/server.go:986 +0xbb
2024-05-31T08:49:52Z logger.go:66: created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 55
2024-05-31T08:49:52Z logger.go:66: 	google.golang.org/grpc@v1.58.3/server.go:997 +0x145
2024-05-31T08:49:52Z logger.go:66: 
2024-05-31T08:49:52Z logger.go:66: Error: The terraform-provider-azurerm_v3.106.0_x5 plugin crashed!
2024-05-31T08:49:52Z logger.go:66: 
2024-05-31T08:49:52Z logger.go:66: This is always indicative of a bug within the plugin. It would be immensely
2024-05-31T08:49:52Z logger.go:66: helpful if you could report the crash with the plugin's maintainers so that it
2024-05-31T08:49:52Z logger.go:66: can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

Lookup of AKS resource should be successful

### Actual Behaviour

Plugin crashes attempting the data lookup

### Steps to Reproduce

Terraform plan against already deployed devices with configuration containing:

```hcl
data "azurerm_kubernetes_cluster" "aks" {
  name                = var.aks_name
  resource_group_name = var.aks_resource_group_name
}
```

Note: pinning provider back to 3.105 works

Interesting behaviour, I've pinned the provider to 3.105 and executed a `terraform destroy` against the deployed resources (this doesn't destroy the cluster itself, that's in a different code base). Now when attempting to deploy completely fresh on 3.106 the data lookups for `azurerm_kuberenetes_cluster`, `azurerm_key_vault` and `azurerm_subnet` exhibit a crash (the cluster, KV and subnet do indeed exist) -  whereas when 3.105 had been used for an `apply`, a `plan` under 3.106 only encountered a crash for the AKS data lookup.

Further interesting, the 3.106 plan has shifting behaviour, doing a plan the first time the kubernetes and subnet lookups fail:

```txt
Error: Plugin did not respond

  with data.azurerm_kubernetes_cluster.aks,
  on data.tf line 1, in data "azurerm_kubernetes_cluster" "aks":
   1: data "azurerm_kubernetes_cluster" "aks" {

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more
details.

Error: Plugin did not respond

  with data.azurerm_subnet.subnet,
  on data.tf line 72, in data "azurerm_subnet" "subnet":
  72: data "azurerm_subnet" "subnet" {

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more
details.

Stack trace from the terraform-provider-azurerm_v3.106.0_x5 plugin:
```

Then attempting another plan on 3.106 again without changing anything shows the subnet lookup succeeding and the kuberenetes lookup continuing to fail:

```txt
2024-05-31T09:39:49Z logger.go:66: data.azurerm_subnet.subnet: Read complete after 0s

Error: Plugin did not respond

  with data.azurerm_kubernetes_cluster.aks,
  on data.tf line 1, in data "azurerm_kubernetes_cluster" "aks":
   1: data "azurerm_kubernetes_cluster" "aks" {

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more
details.

```

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/containers/kubernetes_cluster_data_source.go
+++ b/internal/services/containers/kubernetes_cluster_data_source.go
@@ -1479,7 +1479,7 @@ func flattenKubernetesClusterDataSourceUpgradeSettings(input *managedclusters.Ag
 		values["drain_timeout_in_minutes"] = *input.DrainTimeoutInMinutes
 	}
 
-	if input.DrainTimeoutInMinutes != nil {
+	if input.NodeSoakDurationInMinutes != nil {
 		values["node_soak_duration_in_minutes"] = *input.NodeSoakDurationInMinutes
 	}
 

